src/components/radio.tsx

```diff
@@ -29,16 +29,17 @@ const Radio: FC<RadioProps> = ({ id, label, options, updateValue, value }) => {
         <>
             <Card className="input-card">
                 <Card.Body>
-                    {label}
+                    <label className="mb-3 custom-label">{label}</label>
                     <Form.Group className="mb-3" controlId={label}>
-                        {options.map(option => (
+                        {options.map((option, index) => (
                             <Form.Check
                                 type="radio"
                                 label={option}
                                 name={label}
                                 value={option}
                                 checked={option === value}
                                 onChange={handleOptionChange}
+                                key={index}
```

Comment:
I suggest setting the key to `option` instead. Then `index` is not needed.

Reply:
updated as per the suggestion